ScrapeRepos.ts

```diff
@@ -0,0 +1,356 @@
+          totalCount
+          edges {
+            starredAt
+            node {
+              login
+            }
+          }
+          pageInfo {
+            endCursor
+            hasNextPage
+          }
+        }
+      }
+    }
+  `;
+
+  const headers = {
+    'Authorization': authToken
+  };
+
+  let hasNextPage = true;
+  let endCursor: string | null = null;
+
+  let starHistoryList: any[] = []
+
+  try {
+    while (hasNextPage) {
+      const variables: any = {
+        owner,
+        repo,
+        cursor: endCursor
+      };
+
+      const response = await axios.post(
+        'https://api.github.com/graphql',
+        { query, variables },
+        { headers }
+      );
+      const starsData = response.data.data.repository.stargazers.edges;
+      starHistoryList.push(starsData)
+
+      const pageInfo = response.data.data.repository.stargazers.pageInfo;
+      hasNextPage = pageInfo.hasNextPage;
+      endCursor = pageInfo.endCursor || null;
+
+      variables.cursor = endCursor;
+    }
+  } catch (error) {
+    console.error('Error:', error);
+    return []
+  }
+
+  // get the real star history
+  const mergedArray = [].concat(...starHistoryList);
+  const result = await starHistory(mergedArray);
+  return result
+}
+  
+
+/** Transforms an array of all the logged stars timestamps into a merged array with the star History
+ * @param {[starredAt: 'TIME', node: {Object} ]} starsData Array with the timestamps of the star history
+ * @returns {[date: 'DATE', count: 'COUNT']}:  { date: '2023-05-13', count: 515 }
+ */
+async function starHistory(starsData: any[]) {
+  const dateCounts: { [date: string]: number } = {};
+  starsData.forEach((item) => {
+    const date = item.starredAt.split('T')[0]; // Extract date portion
+    dateCounts[date] = (dateCounts[date] || 0) + 1; // Increment count for the date
+  });
+
+  // Convert dateCounts object into an array of objects
+  const result = Object.keys(dateCounts).map((date) => ({
+    date,
+    count: dateCounts[date]
+  }));
+
+  return result;
+}  
+
+
+/** Get trending developers (and their trending repos) from the github page
+ * @param {number} userInput describes the timeframe; 0: daily, 1: weekly; 2: monthly
+ * @returns   list of {name: 'NAME', username: 'USERNAME', repo: 'REPO'}
+ */
+async function fetchDevelopers(userInput: number){
+  const timeModes = ['daily', 'weekly', 'monthly'];
+
+  interface Developer {
+    name: string;
+    username: string |undefined;
+  }
+
+  interface DeveloperRepo {
+    username: string |undefined;
+    repo: string;
+  }
+
+  axios.get('https://github.com/trending/developers?since=' + timeModes[userInput]).then((response: { data: any; }) => {
+    const html = response.data;
+    const $ = cheerio.load(html);
+    const developers: Developer[] = [];
+    const developerRepos: DeveloperRepo[] = []
+
+    // get the developer names and usernames
+    $('h1.h3.lh-condensed a').each((i: number, el: cheerio.Element) => {
```

Comment:
Please don't use a static html tag, you should use a class which is not used generally like bootstrap classes, or id